call.go

```diff
@@ -61,7 +61,7 @@ func recvResponse(ctx context.Context, dopts dialOptions, t transport.ClientTran
 		if c.maxReceiveMessageSize == nil {
 			return Errorf(codes.Internal, "callInfo maxReceiveMessageSize field uninitialized(nil)")
 		}
-		if err = recv(p, dopts.codec, stream, dopts.dc, reply, *c.maxReceiveMessageSize, inPayload); err != nil {
+		if err = recv(p, dopts.codec, stream, dopts.dc, reply, *c.maxReceiveMessageSize, inPayload, encoding.GetCompressor(c.compressorType)); err != nil {
```

Comment:
Not Done. checkRecvPayload() in the recv() won't ignore it and returns `"compressor not implement"` error. 
In the encode() part to do the same thing before encode.

Do you mean check it and return the error when calling `UseCompressor`? So it can return the error as early as possible?